src/ape_run/_cli.py

```diff
@@ -114,14 +134,28 @@ def commands(self) -> Dict[str, Union[click.Command, click.Group]]:
         if not self._project.scripts_folder.exists():
             return {}
 
+        return self._get_cli_commands(self._project.scripts_folder)
+
+    def _get_cli_commands(self, base_path: Path) -> Dict:
         commands = {}
-        for filepath in self._project.scripts_folder.glob("*.py"):
+
+        for filepath in base_path.iterdir():
             if filepath.stem.startswith("_"):
                 continue  # Ignore any "private" files
 
-            cmd = self._get_command(filepath)
-            if cmd:  # NOTE: Don't allow calling commands that failed to load
-                commands[filepath.stem] = cmd
+            elif filepath.is_dir():
+                group = click.Group(
+                    name=filepath.stem, short_help=f"Run a script from '{filepath.stem}'"
+                )
+                subcommands = self._get_cli_commands(filepath)
+                for subcommand in subcommands.values():
+                    group.add_command(subcommand)
+                commands[filepath.stem] = group
+
+            if filepath.suffix == ".py":
+                cmd = self._get_command(filepath)
+                if cmd:  # NOTE: Don't allow calling commands that failed to load
+                    commands[filepath.stem] = cmd
```

Comment:
![Screenshot 2023-01-17 at 7 54 41 PM](https://user-images.githubusercontent.com/32708219/213062658-f11ef8ba-5bdf-4944-a6d1-d555eeb20087.png)

Looks like mypy is unhappy in the test because of the types here, you're setting the value in the dictionary as a click group in line 153 but as an individual command in line 158

Reply:
A click group is a subtype of Command I believe, so maybe there is a baseclass we can use